--- a/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
+++ b/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
@@ -106,7 +_,7 @@
          List<Entity> list = p_155911_.m_45933_((Entity)null, PistonMath.m_60328_(aabb, direction, d0).m_82367_(aabb));
          if (!list.isEmpty()) {
             List<AABB> list1 = voxelshape.m_83299_();
-            boolean flag = p_155914_.f_60334_.m_60713_(Blocks.f_50374_);
+            boolean flag = p_155914_.f_60334_.isSlimeBlock(); //TODO: is this patch really needed the logic of the original seems sound revisit later
             Iterator iterator = list.iterator();
 
             while(true) {
@@ -280,7 +_,7 @@
             if (p_155916_.m_8055_(p_155917_).m_60713_(Blocks.f_50110_)) {
                BlockState blockstate = Block.m_49931_(p_155919_.f_60334_, p_155916_, p_155917_);
                if (blockstate.m_60795_()) {
-                  p_155916_.m_7731_(p_155917_, p_155919_.f_60334_, 84);
+                  p_155916_.m_7731_(p_155917_, p_155919_.f_60334_, (84 | Block.f_152394_));
                   Block.m_49902_(p_155919_.f_60334_, blockstate, p_155916_, p_155917_, 3);
                } else {
                   if (blockstate.m_61138_(BlockStateProperties.f_61362_) && blockstate.m_61143_(BlockStateProperties.f_61362_)) {
